pkgs/applications/networking/syncthing/default.nix

```diff
@@ -1,57 +1,99 @@
-{ stdenv, lib, fetchFromGitHub, go, procps, removeReferencesTo }:
+{ stdenv, lib, go, procps, removeReferencesTo, fetchFromGitHub }:
 
-stdenv.mkDerivation rec {
-  version = "0.14.44";
-  name = "syncthing-${version}";
+let
+  common =
+    args@{ name, target, installPhase }:
 
-  src = fetchFromGitHub {
-    owner  = "syncthing";
-    repo   = "syncthing";
-    rev    = "v${version}";
-    sha256 = "1gdkx6lbzmdz2hqc9slbq41rwgkxmdisnj0iywx4mppmc2b4v6wh";
-  };
+    stdenv.mkDerivation rec {
+      version = "0.14.44";
+      name = "${args.name}-${version}";
```

Comment:
You can just use ```name``` instead of ```args.name```.

Reply:
I wasn't able to do this, since I got an infinite recursion error (presumably with the value of the `name` attribute itself); instead, I renamed the parameter.